packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/Metric.test.tsx

```diff
@@ -0,0 +1,234 @@
+import type { AvailableMetrics } from '@linode/api-v4';
+const mockData: AvailableMetrics[] = [
+  {
+    available_aggregate_functions: ['min', 'max', 'avg'],
+    dimensions: [
+      {
+        dimension_label: 'cpu',
+        label: 'CPU name',
+        values: [],
+      },
+      {
+        dimension_label: 'state',
+        label: 'State of CPU',
+        values: [
+          'user',
+          'system',
+          'idle',
+          'interrupt',
+          'nice',
+          'softirq',
+          'steal',
+          'wait',
+        ],
+      },
+      {
+        dimension_label: 'LINODE_ID',
+        label: 'Linode ID',
+        values: [],
+      },
+    ],
+    label: 'CPU utilization',
+    metric: 'system_cpu_utilization_percent',
+    metric_type: 'gauge',
+    scrape_interval: '2m',
+    unit: 'percent',
+  },
+];
+
+describe('Metric component tests', () => {
+  const user = userEvent.setup();
+  it('should render all the components and names', () => {
+    const container = renderWithThemeAndHookFormContext<CreateAlertDefinitionForm>(
+      {
+        component: (
+          <Metric
+            apiError={[false, false]}
+            data={mockData}
+            name={`rule_criteria.rules.${0}`}
+            onMetricDelete={vi.fn()}
+            showDeleteIcon={false}
+          />
+        ),
+        useFormOptions: {
+          defaultValues: {
+            serviceType: 'linode',
+          },
+        },
+      }
+    );
+    expect(container.getAllByLabelText('Data Field'));
+    expect(container.getAllByLabelText('Aggregation Type'));
+    expect(container.getAllByLabelText('Operator'));
+    expect(container.getAllByLabelText('Threshold'));
+  });
+
+  it('should render the Data Field component with options happy path and select an option', async () => {
+    const container = renderWithThemeAndHookFormContext<CreateAlertDefinitionForm>(
+      {
+        component: (
+          <Metric
+            apiError={[false, false]}
+            data={mockData}
+            name={`rule_criteria.rules.${0}`}
+            onMetricDelete={vi.fn()}
+            showDeleteIcon={false}
+          />
+        ),
+        useFormOptions: {
+          defaultValues: {
+            serviceType: 'linode',
+          },
+        },
+      }
+    );
+    const dataFieldContainer = container.getByTestId('Data-field');
+    expect(
+      within(dataFieldContainer).getByRole('button', {
+        name:
+          'Represents the metric you want to receive alerts for. Choose the one that helps you evaluate performance of your service in the most efficient way.',
+      })
+    );
+    const dataFieldInput = within(dataFieldContainer).getByRole('button', {
+      name: 'Open',
+    });
+    user.click(dataFieldInput);
+
+    expect(
+      await container.findByRole('option', { name: mockData[0].label })
+    ).toBeInTheDocument();
+
+    await user.click(
+      container.getByRole('option', { name: mockData[0].label })
```

Comment:
is await needed for this , since it is using mockdata, earlier it came as comment from CM team, hence asking

Reply:
The await here is because of the user.click, for userEvent we are using await. 
https://github.com/linode/manager/blob/1072d546d7165e4a59aaaa753311a24a100cac50/packages/manager/src/features/CloudPulse/Alerts/CreateAlert/GeneralInformation/EngineOption.test.tsx#L31-L32